### PR TITLE
Add application structure guidelines

### PR DESCRIPTION
**Pull request by agnivade**
#### Summary

Add application structure guidelines

This is something which I feel needs to be mentioned and should be helpful. A lot of repos are being created, each with the developer's own taste. Some amount of unification will be helpful.

I have refrained from being too preachy. Just kept it to basic common sense and some anti-patterns to avoid. Rest is up to the developer.

### CODE REVIEW

**Review by isacikgoz — APPROVED** [review]
This is great @agnivade, if you think adding a couple of resources for further reading; I have two suggestions for that
- https://medium.com/@benbjohnson/standard-package-layout-7cdbc8391fc1
- https://eli.thegreenplace.net/2019/simple-go-project-layout-with-modules/

**Comment by agnivade** [comment]
Yeah, I love Ben's guide. But I have refrained from adding any links because this is an extremely opinionated topic.

What I have basically done is to say that don't try to apply any pattern right off the bat. Let the project grow, understand the requirements, then split into packages. And avoid `pkg`, `utils`. That's it. Anything else is up to the developer to do.

If you want, I can add some more lines mentioning that there is a lot of material that explains how to setup a Go project, however this is a basic guideline which should be followed before doing any of those.

**Comment by isacikgoz** [comment]
I see, let's leave it as is then.

**Review by grundleborg — APPROVED** [review]
LGTM

File: site/content/contribute/server/style-guide.md [hunk 1]
```diff
@@ -27,6 +27,33 @@ This does not, however, mean that a developer is *required* to fix any surroundi
...
+
+Putting everything under `internal` has the advantage that you are free to change whatever you want without any fallout. After some time, when the project grows, package boundaries will start to appear. At that point, start to separate into sub-packages.
+
+Following are some of the anti-patterns to keep in mind:
+
+- Don't use the `pkg` pattern. This is a common standard used by many projects. But it goes against the Go philosophy of naming packages that signify what they contain. From https://blog.golang.org/package-names:
+
+>  A package's name provides context for its contents, making it easier for clients to understand what the package is for and how to use it.
+
+The `pkg` directory was used long ago in the Go project when there wasn't any well-defined standards. But it was later removed to just have normal packages.
```
Comment (justinegeffen) [inline site/content/contribute/server/style-guide.md:51]:
```suggestion
The `pkg` directory was used long ago in the Go project when there weren't any well-defined standards. But it was later removed to just have normal packages.
```

**Review by justinegeffen — APPROVED** [review]

**Pull request merged**